coordinator/cluster/cluster.go

```diff
@@ -127,6 +162,13 @@ func (c *LoadAgentCluster) Shutdown() {
 			}
 		}(ag)
 	}
+
+	for _, browserAgent := range c.browserAgents {
+		if _, err := browserAgent.Destroy(); err != nil {
+			c.log.Error("cluster: failed to stop browser agent", mlog.Err(err))
+		}
+	}
```

Comment:
Something like this (not tested):
```suggestion
	for _, browserAgent := range c.browserAgents {
		go func(ag *client.Agent) {
			dever wg.Done()
			if _, err := browserAgent.Destroy(); err != nil {
				c.log.Error("cluster: failed to stop browser agent", mlog.Err(err))
			}
		}(browserAgent)
	}
```

Reply:
You're absolutely right! There was a bug in the wait group logic. I was adding len(c.agents) + len(c.browserAgents) to the wait group, but the browser agents loop was missing the goroutine and wg.Done() call.